AirCasting/SessionsSynchronization/Adapters/DataStore/SessionSynchronizationDatabase.swift

```diff
@@ -0,0 +1,152 @@
+        }
+    }
+    
+    public func removeSessions(with uuids: [SessionUUID]) -> Future<Void, Error> {
+        Future { promise in
+            self.database.removeSessions(where: .predicate(NSPredicate(format: "uuid IN %@", uuids))) { error in
+                if let error = error {
+                    promise(.failure(error))
+                } else {
+                    promise(.success(()))
+                }
+            }
+        }
+    }
+    
+    public func readSession(with uuid: SessionUUID) -> Future<SessionsSynchronization.SessionStoreSessionData, Error> {
+        Future { promise in
+            self.database.fetchSessions(constrained: .predicate(NSPredicate(format: "uuid == %@", uuid.rawValue)), completion: { result in
+                switch result {
+                case .failure(let error):
+                    promise(.failure(error))
+                case .success(let entities):
+                    guard let sessionEntity = entities.first else {
+                        promise(.failure(SessionSynchronizationDatabaseError.sessionNotFound))
+                        return
+                    }
+                    promise(.success(sessionEntity.toSynchronizationStoreData()))
+                }
+            })
+        }
+    }
+}
+
+extension Database.Session {
+    func toSynchronizationStoreData() -> SessionsSynchronization.SessionStoreSessionData {
+        let measurements = measurementStreams?.map { stream -> SessionsSynchronization.SessionStoreMeasurementStreamData in
+            // TODO: Are those force unwraps safe here?
+            return SessionsSynchronization.SessionStoreMeasurementStreamData(id: stream.id!,
+                                                                             measurementShortType: stream.measurementShortType!,
+                                                                             measurementType: stream.measurementType!,
+                                                                             sensorName: stream.sensorName!,
+                                                                             sensorPackageName: stream.sensorPackageName!,
+                                                                             thresholdHigh: Int(stream.thresholdHigh),
+                                                                             thresholdLow: Int(stream.thresholdLow),
+                                                                             thresholdMedium: Int(stream.thresholdMedium),
+                                                                             thresholdVeryHigh: Int(stream.thresholdVeryHigh),
+                                                                             thresholdVeryLow: Int(stream.thresholdVeryLow),
+                                                                             unitName: stream.unitName!,
+                                                                             unitSymbol: stream.unitSymbol!)
```

Comment:
The new class for `Database <-> SessionSyncStore` translation will handle this. But it's quite annoying to deal with CoreData optionals, we should make a list of stuff that is **guaranteed** or smth and make it so that the `Database` layer structs like `Database.MeasurementsStream` have all optionality sorted out correctly. This way the rest of the app can breathe a bit and not worry about this issue at all

Reply:
Well it fitted quite nicely into the existing `SynchronizationDataConverter`. But those bangs are something to talk about, I don't want to pass optionals everywhere and the only reason we have them is `CoreData` which is a technical detail, we should make some rules and not let this overwhelm us with optionals.